### PR TITLE
Integrate RTL CSS generation in the core

### PR DESCRIPTION
**Pull request by eternoendless**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.3.x
| Description?  | Moved RTL generation from an external dependency into the core while keeping CSSJanus, and added some minor improvements from this other PR: https://github.com/PrestaShop/PrestaShop/pull/8642
| Type?         | improvement
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | n/a
| How to test?  | Generating an RTL language should work the same as before

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/8657)

### CODE REVIEW

File: src/Core/Localization/RTL/StylesheetGenerator.php [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Localization\RTL;
+
+use PrestaShop\PrestaShop\Core\Localization\RTL\Exception\GenerationException;
+use \CSSJanus;
+use \Tools;
```
Comment (Quetzacoalt91) [inline src/Core/Localization/RTL/StylesheetGenerator.php:31]: No need to write the first slash with `use`

File: composer.json [hunk 2]
```diff
@@ -97,8 +96,15 @@
...
         "composer/ca-bundle": "^1.0",
         "nikic/php-parser": "^2.1",
-        "prestashop/decimal": "^1.0.0"
+        "prestashop/decimal": "^1.0.0",
+        "cssjanus/cssjanus": "dev-patch-1"
     },
+    "repositories": [
+        {
+            "type": "vcs",
+            "url": "https://github.com/prestashop/php-cssjanus"
```
Comment (Quetzacoalt91) [inline composer.json:105]: Can you please describe what is the reason of that fork?

**Review by Quetzacoalt91 — APPROVED** [review]
Looks good to me.

For the next time, you should rely on the libraries we already use on the project. For instance, finding, reading and writing files can be done with the `Finder` & `Filesystem` from symfony.

File: composer.json (hunk 2 above)
Comment (eternoendless) [reply composer.json:105]: Long story short -- it doesn't handle some CSS files. You can learn more here: https://github.com/cssjanus/php-cssjanus/pull/13

The idea is to come back to the original library if/when the maintainer decides to merge that PR.

File: src/Core/Localization/RTL/StylesheetGenerator.php (hunk 1 above)
Comment (eternoendless) [reply src/Core/Localization/RTL/StylesheetGenerator.php:31]: The More You Know

**Comment by eternoendless** [comment]
Yeah, I thought about it but decided to stick with the same tools used in the first place. We can improve on that later.

**Pull request merged**